concourse/concourse
# web: archived pipeline toggle switch

## What does this PR accomplish?

Bug Fix | **Feature** | Documentation

closes #5640.
closes #5635.

## Changes proposed by this PR:

* "show archived" toggle switch for dashboard
  * Introduce Toggle module for rendering toggle switches
  * Add ?view=all query param to dashboard for rendering archived pipelines
* Change drag and drop to be relative to pipeline names (rather than list indexes)
* Simplify pipeline cards for archived pipelines (as per #5635)

## Notes to reviewer:


* To test drag and drop:
  1. `fly -t dev set-pipeline -p p1 -c pipeline.yml`
  1. `fly -t dev set-pipeline -p p2 -c pipeline.yml`
  1. `fly -t dev set-pipeline -p p3 -c pipeline.yml`
  1. `fly -t dev archive-pipeline -p p2`
  1. With the toggle switch set in its "off" state, drag and drop `p3` before `p1`
  1. On master, this won't work - `p3` will appear after `p1` after dropping

## Contributor Checklist

- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [x] Added tests (Unit and/or Integration)
- [ ] Updated [Release notes]

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs
[Release notes]: https://github.com/concourse/concourse/tree/master/release-notes

## Reviewer Checklist

- [x] Tests reviewed
- [x] Release notes reviewed
- [x] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).
